static/js/hoc/withNavSidebar.js

```diff
@@ -1,30 +1,8 @@
 // @flow
-import React from "react"
+import NavSidebar from "../components/NavSidebar"
+import withSidebars from "../hoc/withSidebars"
 
-import Sidebar from "../components/Sidebar"
-import Navigation from "../components/Navigation"
+export const withNavSidebar = withSidebars(NavSidebar)
 
-const withNavSidebar = (WrappedComponent: Class<React.Component<*, *>>) => {
-  class WithNavSidebar extends React.Component<*, *> {
-    render() {
-      const { subscribedChannels, location: { pathname } } = this.props
-
-      return (
-        <div className="content">
-          <Sidebar>
-            <Navigation
-              subscribedChannels={subscribedChannels}
-              pathname={pathname}
-            />
-          </Sidebar>
-          <div className="main-content">
-            <WrappedComponent {...this.props} />
-          </div>
-        </div>
-      )
-    }
-  }
-  return WithNavSidebar
-}
-
-export default withNavSidebar
```

Comment:
I might suggest we simplify this a little bit and just declare two HOCs (`withNavSidebar` and `withNavAndChannelSidebars`) that just take a `WrappedComponent` and render stuff directly - as is there's kind of a lot of indirection going on that I don't quite see the need for. Figuring out what `withNavSidebar = withSidebars(NavSidebar)` means, for instance, requires quite a bit of poking around. My thinking is that right now we only have two variations (nav and nav + channel description) and I don't think we're likely to have more anytime soon.